### PR TITLE
✨  add strelka2 module

### PR DESCRIPTION
**Pull request by dmiller15**
# Pull Request Template

## Description

Review everything. Adds strelka2 module.

Part of https://github.com/d3b-center/bixu-tracker/issues/1337

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] https://cavatica.sbgenomics.com/u/d3b-bixu/dev-canine-workflow/tasks/d1cdaaac-697f-4cf5-bacb-2e0c2764b351/#

**Test Configuration**:
* Environment:
* Test files:

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: subworkflows/canine_strelka2_module.cwl [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+      ram: strelka2_ram
+    out: [indels, snvs, realigned_normal_bam, realigned_tumor_bam]
+
+  samtools_view_normal_cram:
+    run: ../tools/samtools_view.cwl
+    when: $(inputs.input_reads != null)
+    in:
+      input_reads: strelka2_somatic/realigned_normal_bam
+      output_cram:
+        source: disable_workflow # hiding this here because I hate cavatica
```
Comment (sakshamphul) [inline subworkflows/canine_strelka2_module.cwl:63]: haha

File: tools/samtools_view.cwl [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  views and converts SAM/BAM/CRAM files
+requirements:
+  - class: ShellCommandRequirement
+  - class: InlineJavascriptRequirement
+  - class: DockerRequirement
+    dockerPull: 'pgc-images.sbgenomics.com/d3b-bixu/samtools:1.15.1'
+  - class: ResourceRequirement
+    ramMin: $(inputs.ram*1000)
+    coresMin: $(inputs.cpu)
+baseCommand: []
```
Comment (sakshamphul) [inline tools/samtools_view.cwl:14]: Is it necessary to have _baseCommand_ if we do not have command line to run ?

File: tools/samtools_view.cwl [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    type: 'int?'
+    default: 4
+    doc: "GB size of RAM to allocate to this task."
+
+outputs:
+  output:
+    type: File
+    secondaryFiles: [{pattern: '.bai', required: false}, {pattern: '^.bai', required: false}, {pattern: '.crai', required: false}, {pattern: '^.crai', required: false}]
+    outputBinding:
+      glob: '*.*am'
```
Comment (sakshamphul) [inline tools/samtools_view.cwl:86]: is this for *.bam and/or *.cram? Would this work for both?

**Review by sakshamphul — APPROVED** [review]

File: tools/samtools_view.cwl (hunk 2 above)
Comment (dmiller15) [reply tools/samtools_view.cwl:14]: By spec definition, it is not required. I just keep it around for completeness.

File: tools/samtools_view.cwl (hunk 3 above)
Comment (dmiller15) [reply tools/samtools_view.cwl:86]: Yes, this will grab SAMs, BAMs, or CRAMs

**Review by migbro — APPROVED** [review]

**Pull request merged**